iOS/Collection View Management /CollectionViewDataSource.swift

```diff
@@ -19,11 +19,12 @@ class CollectionViewDataSource<SectionType: Hashable, ItemType: Hashable> {
 
     private var diffableDataSource: DiffableDataSourceType!
 
-    var didSelectItem: ((ItemType) -> Void)? 
+    var didSelectItem: ((ItemType) -> Void)?
 
     required init(collectionView: UICollectionView) {
         self.diffableDataSource = DiffableDataSourceType(collectionView: collectionView,
-                                                         cellProvider: { collectionView, indexPath, itemIdentifier in
+                                                         cellProvider: {
+            [unowned self] (collectionView, indexPath, itemIdentifier) in
```

Comment:
Just needed an unowned reference in the cell providers. This was affecting all subclasses of the collectionview datasource.

Reply:
🙈